base/dict.jl

```diff
@@ -8,7 +8,10 @@ const secret_table_token = :__c782dbf1cf4d6a2e5e3865d7e95634f2e09b5902__
 
 haskey(d::Associative, k) = in(k,keys(d))
 
-function in(p::Tuple{Any,Any}, a::Associative)
+function in(p, a::Associative)
+    if !isa(p,Pair)
+        error("Associative collections only contain Pairs; look for e.g. A=>B instead")
```

Comment:
We could use dispatch, but there is a fallback definition of `in` for all types that will unhelpfully just return `false` in this case. This isn't really a type error, since it's perfectly valid to ask whether some random thing is in a collection. This is kind of a "fuzzy" error just meant to be helpful.